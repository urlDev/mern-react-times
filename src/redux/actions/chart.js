import axios from "axios";

export const FETCH_FOREX_SUCCESS = "FETCH_FOREX_SUCCESS";
export const FETCH_CHART_ERROR = "FETCH_CHART_ERROR";
export const SET_LOADING = "SET_LOADING";
export const SET_LOADING_TRUE = "SET_LOADING_TRUE";
export const CHANGE_MARKET_TYPE = "CHANGE_MARKET_TYPE";
export const CHANGE_MARKET_NAME = "CHANGE_MARKET_NAME";
export const SET_MARKET_DETAIL = "SET_MARKET_DETAIL";
export const SET_CHART_DATA = "SET_CHART_DATA";
export const CLEAN_CHART_DATA = "CLEAN_CHART_DATA";
export const SET_CHART_TIME_FRAME = "SET_CHART_TIME_FRAME";
export const SET_RATING = "SET_RATING";
export const SET_HOME_CHART_DATA = "SET_HOME_CHART_DATA";
export const CLEAN_STATE = "CLEAN_STATE";
export const GET_SEARCH_RESULTS = "GET_SEARCH_RESULTS";
export const SET_SEARCH_MARKET_DETAIL = "SET_SEARCH_MARKET_DETAIL";
export const CLEAR_SEARCH_RESULTS = "CLEAR_SEARCH_RESULTS";
export const OPEN_SEARCH_MODAL = "OPEN_SEARCH_MODAL";
export const CLOSE_SEARCH_MODAL = "CLOSE_SEARCH_NODAL";

export const closeSearchModal = () => ({
  type: CLOSE_SEARCH_MODAL,
});

export const openSearchModal = () => ({
  type: OPEN_SEARCH_MODAL,
});

export const clearSearchResults = () => ({
  type: CLEAR_SEARCH_RESULTS,
});

export const setSearchMarketDetail = (details) => ({
  type: SET_SEARCH_MARKET_DETAIL,
  payload: details,
});

export const getSearchResults = (result) => ({
  type: GET_SEARCH_RESULTS,
  payload: result,
});

export const setLoading = () => ({
  type: SET_LOADING,
});

export const setLoadingTrue = () => ({
  type: SET_LOADING_TRUE,
});

export const cleanState = () => ({
  type: CLEAN_STATE,
});

export const setHomeChartData = (chart) => ({
  type: SET_HOME_CHART_DATA,
  payload: [chart],
});

export const setRating = (rating) => ({
  type: SET_RATING,
  payload: rating,
});

export const setChartTimeFrame = (time) => ({
  type: SET_CHART_TIME_FRAME,
  payload: time,
});

export const setChartData = (data) => ({
  type: SET_CHART_DATA,
  payload: data,
});

export const cleanChartData = () => ({
  type: CLEAN_CHART_DATA,
});

export const setMarketDetail = (market) => ({
  type: SET_MARKET_DETAIL,
  payload: market,
});

export const changeMarketType = (market) => ({
  type: CHANGE_MARKET_TYPE,
  payload: market,
});

export const changeMarketName = (market) => ({
  type: CHANGE_MARKET_NAME,
  payload: market,
});

export const fetchForexSuccess = (chart) => ({
  type: FETCH_FOREX_SUCCESS,
  payload: chart,
});

export const fetchChartError = (error) => ({
  type: FETCH_CHART_ERROR,
  payload: error,
});

export const fetchForex = (market) => async (dispatch) => {
  try {
    const response = await axios.get(
      `https://financialmodelingprep.com/api/v3/quote/${market}?apikey=${process.env.REACT_APP_CHART_KEY}`
    );
    const data = await response.data;
    // FinancialModel API doesn't send error if it's about limit reach.
    // In that sense, they send an error message with 200 status. Weird.
    // Thats why I am catching that error here and dispatching it.
    if (data.length) {
      return data.length === 1
        ? [dispatch(setSearchMarketDetail(data))]
        : dispatch(fetchForexSuccess(data));
    } else {
      return dispatch(fetchChartError(data));
    }
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};

export const fetchChartData = (symbol, timeFrame) => async (dispatch) => {
  dispatch(cleanChartData());
  try {
    const response = await axios.get(
      `https://financialmodelingprep.com/api/v3/historical-chart/${timeFrame}/${symbol}?apikey=${process.env.REACT_APP_CHART_KEY}`
    );
    const data = await response.data;
    return dispatch(setChartData(data));
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};

export const fetchRating = (symbol) => async (dispatch) => {
  try {
    const response = await axios.get(
      `https://financialmodelingprep.com/api/v3/rating/${symbol}?apikey=${process.env.REACT_APP_CHART_KEY}`
    );
    const data = await response.data;
    return dispatch(setRating(data));
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};

/*
This deserves to be mentioned.
Few months ago, when I was making the first version of this app, I used array.map to fetch charts.
Because of the async nation and using array map method, response was coming irregular and I had to sort them.
Now, with using for of loop, I dont need to write any extra code because loop waits for one to finish to move on.
Plus, its so clean!
Yet another homerun for me! 😁😎
*/
export const fetchHomeChart = (symbols) => async (dispatch) => {
  const symbolsArray = symbols.split(",").sort();
  let charts = [];
  try {
    // dispatch(setLoadingTrue());
    dispatch(cleanState());
    for (let symbol of symbolsArray) {
      const response = await axios.get(
        `https://financialmodelingprep.com/api/v3/historical-chart/1hour/${symbol}?apikey=${process.env.REACT_APP_CHART_KEY}`
      );
      const data = response.data;
      charts.push(data);
    }
    return charts.length
      ? dispatch(setHomeChartData(charts))
      : dispatch(fetchChartError(charts));
    // dispatch(setLoading());
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};

export const fetchSearch = (input) => async (dispatch) => {
  try {
    const response = await axios.get(
      `https://financialmodelingprep.com/api/v3/search?query=${input}&limit=10&apikey=${process.env.REACT_APP_CHART_KEY}`
    );
    const data = response.data;
    return dispatch(getSearchResults(data));
  } catch (error) {
    return dispatch(fetchChartError(error));
  }
};
