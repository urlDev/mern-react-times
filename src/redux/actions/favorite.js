import axios from "axios";
import toaster from "toasted-notes";

import NotificationComponent from "../../components/notification-component/NotificationComponent";

const url = "https://urldev-mern-react-times-api.herokuapp.com";
// const url = `http://localhost:3000`;

export const GET_FAVORITE = "GET_FAVORITE";
export const ADD_FAVORITE = "ADD_FAVORITE";
export const DELETE_FAVORITE = "DELETE_FAVORITE";
export const FETCH_FAVORITE_ERROR = "FETCH_FAVORITE_ERROR";
export const CLEAN_FAVORITE_STATE = "CLEAN_FAVORITE_STATE";

export const getFavorite = (stock) => ({
  type: GET_FAVORITE,
  payload: stock,
});

export const addFavorite = (stock) => ({
  type: ADD_FAVORITE,
  payload: stock,
});

export const deleteFavorite = (stock) => ({
  type: DELETE_FAVORITE,
  payload: stock,
});

export const fetchFavoriteError = (error) => ({
  type: FETCH_FAVORITE_ERROR,
  payload: error,
});

export const cleanFavoriteState = () => ({
  type: CLEAN_FAVORITE_STATE,
});

export const fetchGetFavorites = () => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem("token"));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };
  try {
    const response = await axios.get(`${url}/`, config);
    const data = await response.data;
    return dispatch(getFavorite(data));
  } catch (error) {
    return dispatch(fetchFavoriteError(error));
  }
};

export const fetchAddFavorites = (stock) => async (dispatch) => {
  const favoriteStock = {
    symbol: stock,
  };

  const token = JSON.parse(localStorage.getItem("token"));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };

  try {
    const response = await axios.post(`${url}/`, favoriteStock, config);
    const data = await response.data;
    return [
      dispatch(addFavorite(data.symbol[0])),
      toaster.notify(
        () => (
          <NotificationComponent
            success={true}
            text={`You added ${data.symbol[0].name
              .split("^")
              .join("")} to your favorites successfully.`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(fetchFavoriteError(error)),
      toaster.notify(
        () => (
          <NotificationComponent
            success={false}
            text={`We just broke something, try again please or refresh.`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

export const fetchDeleteFavorite = (id) => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem("token"));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };
  try {
    const response = await axios.delete(`${url}/${id}`, config);
    const data = await response.data;
    return [
      dispatch(deleteFavorite(data)),
      toaster.notify(
        () => (
          <NotificationComponent
            success={true}
            text={`You deleted ${data.symbol[0].name
              .split("^")
              .join("")} from your favorites.`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(fetchFavoriteError(error)),
      toaster.notify(
        () => (
          <NotificationComponent
            success={false}
            text={`We just broke something, try again please or refresh.`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};
