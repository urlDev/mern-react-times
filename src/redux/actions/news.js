import axios from 'axios';

export const FETCH_POPULAR_SUCCESS = 'FETCH_POPULAR_SUCCESS';
export const FETCH_STORY_SUCCESS = 'FETCH_STORY_SUCCESS';
export const FETCH_NEWS_ERROR = 'FETCH_NEWS_ERROR';
export const CHANGE_HEADER = 'CHANGE_HEADER';

export const fetchPopularSuccess = (news) => ({
    type: FETCH_POPULAR_SUCCESS,
    payload: news,
});

export const fetchStorySuccess = (news) => ({
    type: FETCH_STORY_SUCCESS,
    payload: news,
});

export const changeHeader = (header) => ({
    type: CHANGE_HEADER,
    payload: header,
});

export const fetchNewsError = (error) => ({
    type: FETCH_NEWS_ERROR,
    payload: error,
});

export const fetchTopStories = (topStories) => async(dispatch) => {
    try {
        const response = await axios.get(
            `https://api.nytimes.com/svc/topstories/v2/${topStories}.json?api-key=${process.env.REACT_APP_NEWS_API_KEY}`,
        );
        const data = await response.data;
        return dispatch(fetchStorySuccess(data));
    } catch (error) {
        return dispatch(fetchNewsError(error));
    }
};

export const fetchMostPopular = () => async(dispatch) => {
    try {
        const response = await axios.get(
            `https://api.nytimes.com/svc/mostpopular/v2/viewed/7.json?api-key=${process.env.REACT_APP_NEWS_API_KEY}`,
        );
        const data = await response.data;
        return dispatch(fetchPopularSuccess(data));
    } catch (error) {
        return dispatch(fetchNewsError(error));
    }
};