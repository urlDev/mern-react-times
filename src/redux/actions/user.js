import axios from 'axios';
import toaster from 'toasted-notes';
import { push } from 'connected-react-router';

import NotificationComponent from '../../components/notification-component/NotificationComponent';

const url = process.env.REACT_APP_API_URL;

export const REGISTER_USER = 'REGISTER_USER';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGIN_USER = 'LOGIN_USER';
export const DELETE_USER = 'DELETE_USER';
export const UPDATE_USER = 'UPDATE_USER';
export const GET_USER = 'GET_USER';
export const SET_TOKEN = 'SET_TOKEN';
export const USER_FETCH_ERROR = 'USER_FETCH_ERROR';
export const USER_LOADING = 'USER_LOADING';
export const USER_LOADING_END = 'USER_LOADING_END';
export const USER_MODAL_OPEN = 'USER_MODAL_OPEN';
export const USER_MODAL_CLOSE = 'USER_MODAL_CLOSE';
export const DELETE_MODAL_OPEN = 'DELETE_MODAL_OPEN';
export const DELETE_MODAL_CLOSE = 'DELETE_MODAL_CLOSE';
export const UPLOAD_AVATAR = 'UPLOAD_AVATAR';

export const registerUser = (user) => ({
  type: REGISTER_USER,
  payload: user,
});

export const logOutUser = () => ({
  type: LOGOUT_USER,
});

export const loginUser = (user) => ({
  type: LOGIN_USER,
  payload: user,
});

export const deleteUser = () => ({
  type: DELETE_USER,
});

export const updateUser = (user) => ({
  type: UPDATE_USER,
  payload: user,
});

export const getUser = (user) => ({
  type: GET_USER,
  payload: user,
});

export const setToken = (token) => ({
  type: SET_TOKEN,
  payload: token,
});

export const userFetchError = (error) => ({
  type: USER_FETCH_ERROR,
  payload: error,
});

export const userLoading = () => ({
  type: USER_LOADING,
});

export const userLoadingEnd = () => ({
  type: USER_LOADING_END,
});

export const userModalOpen = () => ({
  type: USER_MODAL_OPEN,
});

export const userModalClose = () => ({
  type: USER_MODAL_CLOSE,
});

export const deleteModalOpen = () => ({
  type: DELETE_MODAL_OPEN,
});

export const deleteModalClose = () => ({
  type: DELETE_MODAL_CLOSE,
});

export const uploadAvatar = (images) => ({
  type: UPLOAD_AVATAR,
  payload: images,
});

export const fetchRegisterUser = (user) => async (dispatch) => {
  dispatch(userLoading());
  try {
    const response = await axios.post(`${url}/profile/register`, user);
    const data = await response.data;
    return [
      dispatch(registerUser(data.user)),
      dispatch(userLoadingEnd()),
      dispatch(setToken(data.token)),
      localStorage.setItem('token', JSON.stringify(data.token)),
      localStorage.setItem('user', JSON.stringify(data.user)),
      dispatch(push('/')),
      toaster.notify(
        () => (
          <NotificationComponent
            success={true}
            text={`Welcome! Enjoy your time`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      dispatch(userLoadingEnd()),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Oops! Something went wrong!'}
            success={false}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

export const fetchLogoutUser = () => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };

  const path = window.location.pathname;

  try {
    await axios.post(`${url}/profile/logout`, null, config);

    return [
      path.includes('profile') && dispatch(push('/home')),
      dispatch(logOutUser()),
      toaster.notify(
        () => <NotificationComponent text={'Buh-Bye!'} success={true} />,
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return dispatch(userFetchError(error));
  }
};

export const fetchLoginUser = (user) => async (dispatch) => {
  dispatch(userLoading());
  try {
    const response = await axios.post(`${url}/profile/login`, user);
    const data = await response.data;
    return [
      dispatch(loginUser(data.user)),
      dispatch(userLoadingEnd()),
      dispatch(setToken(data.token)),
      localStorage.setItem('token', JSON.stringify(data.token)),
      localStorage.setItem('user', JSON.stringify(data.user)),
      dispatch(push('/home')),
      toaster.notify(
        () => (
          <NotificationComponent
            success={true}
            text={`Welcome! Enjoy your time`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      dispatch(userLoadingEnd()),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Oops! Something went wrong!'}
            success={false}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

// To get the user,
// but because I added an or statement in user and token initial states,
// app will check if there is user and token in localStorage
// so there will be no need for this.
// Adding just in case.
export const fetchUser = () => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };
  dispatch(userLoading());
  try {
    const response = await axios.get(`${url}/profile`, config);
    const data = await response.data;
    return [
      dispatch(registerUser(data.user)),
      dispatch(userLoadingEnd()),
      dispatch(setToken(data.token)),
      localStorage.setItem('token', JSON.stringify(data.token)),
      localStorage.setItem('user', JSON.stringify(data.user)),
    ];
  } catch (error) {
    return [dispatch(userFetchError(error)), dispatch(userLoadingEnd())];
  }
};

export const fetchUpdateUser = (user) => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };
  try {
    const response = await axios.patch(`${url}/profile`, user, config);
    const data = await response.data;
    return [
      dispatch(updateUser(data)),
      localStorage.setItem('user', JSON.stringify(data)),
      dispatch(push('/')),
      toaster.notify(
        () => (
          <NotificationComponent
            success={true}
            text={`Updated successfully!`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Oops! Something went wrong!'}
            success={false}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

export const fetchDeleteUser = () => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };
  try {
    await axios.delete(`${url}/profile`, config);
    return [
      localStorage.clear(),
      dispatch(push('/')),
      dispatch(deleteUser()),
      toaster.notify(
        () => <NotificationComponent success={true} text={`Buh-bye!`} />,
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      toaster.notify(
        () => (
          <NotificationComponent
            success={false}
            text={`Something went wrong!`}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};

export const fetchUploadAvatar = (input) => async (dispatch) => {
  const token = JSON.parse(localStorage.getItem('token'));

  const config = {
    headers: { Authorization: `Bearer ${token}` },
  };

  try {
    const response = await axios.post(`${url}/profile/avatar`, input, config);
    const data = await response.data;
    return [
      dispatch(uploadAvatar(data)),
      localStorage.setItem('user', JSON.stringify(data)),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Avatar is changed! Wow, that looks amazing! ;)'}
            success={true}
          />
        ),
        { duration: 1500 }
      ),
    ];
  } catch (error) {
    return [
      dispatch(userFetchError(error)),
      toaster.notify(
        () => (
          <NotificationComponent
            text={'Oops! Something went wrong!'}
            success={false}
          />
        ),
        { duration: 1500 }
      ),
    ];
  }
};
