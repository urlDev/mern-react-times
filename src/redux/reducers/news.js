import {
  FETCH_NEWS_ERROR,
  FETCH_POPULAR_SUCCESS,
  FETCH_STORY_SUCCESS,
  CHANGE_HEADER,
  CLEAN_STATE,
  CLEAR_ERROR,
  SET_WIDTH,
  OPEN_RESPONSIVE_MENU,
  CLOSE_RESPONSIVE_MENU,
} from "../actions/news";

const initialState = {
  loadingNews: true,
  popular: [],
  story: [],
  errorNews: null,
  header: "home",
  width: window.innerWidth,
  responsiveMenu: false,
};

const newsReducer = (state = initialState, action) => {
  switch (action.type) {
    case CLEAR_ERROR:
      return {
        ...state,
        errorNews: null,
      };
    case FETCH_POPULAR_SUCCESS:
      return {
        ...state,
        loadingNews: false,
        popular: action.payload.results.slice(0, 4),
      };
    case FETCH_STORY_SUCCESS:
      return {
        ...state,
        loadingNews: false,
        story: action.payload.results.slice(0, 3),
      };
    case FETCH_NEWS_ERROR:
      return {
        ...state,
        errorNews: action.payload,
      };
    case CHANGE_HEADER:
      return {
        ...state,
        header: action.payload,
      };
    case SET_WIDTH:
      return {
        ...state,
        width: action.payload,
      };
    case OPEN_RESPONSIVE_MENU:
      return {
        ...state,
        responsiveMenu: true,
      };
    case CLOSE_RESPONSIVE_MENU:
      return {
        ...state,
        responsiveMenu: false,
      };
    default:
      return state;
  }
};

export default newsReducer;
