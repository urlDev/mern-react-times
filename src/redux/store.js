import { createStore, applyMiddleware } from 'redux';
import rootReducer from './rootReducer';

import thunk from 'redux-thunk';
import logger from 'redux-logger';

const middlewares = [thunk, logger];

export default createStore(rootReducer, applyMiddleware(...middlewares));