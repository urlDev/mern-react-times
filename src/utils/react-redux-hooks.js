// Mocking redux hooks
import {
  useSelector as originalUseSelector,
  useDispatch as originalUseDispatch,
} from "react-redux";

export const useSelector = (state) => originalUseSelector(state);
export const useDispatch = () => originalUseDispatch();
