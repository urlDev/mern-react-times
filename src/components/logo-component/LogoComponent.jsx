import { useDispatch } from "react-redux";

import { changeHeader } from "../../redux/actions/news";

import { Logo } from "./LogoComponent.styles";

const LogoComponent = ({ logo }) => {
  const dispatch = useDispatch();

  return (
    <Logo to="/home">
      <img
        style={{ height: "16px" }}
        src={logo}
        alt="react times logo"
        onClick={() => dispatch(changeHeader("home"))}
      />
    </Logo>
  );
};

export default LogoComponent;
