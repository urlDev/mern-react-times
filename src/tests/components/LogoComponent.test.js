import { shallow } from "enzyme";

import * as ReactReduxHooks from "../../utils/react-redux-hooks";
import { mockStore } from "../store";
import { CHANGE_HEADER } from "../../redux/actions/news";

import LogoComponent from "../../components/logo-component/LogoComponent";

describe("Testing LogoComponent", () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore({});

    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);

    wrapper = shallow(<LogoComponent store={store} />);
  });

  test("Should render the component successfully", () => {
    expect(wrapper).toHaveLength(1);
  });

  test("Should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should dispatch changeHeader action on click", () => {
    wrapper.find("img").simulate("click");
    const actions = store.getActions();

    expect(actions).toEqual([{ type: CHANGE_HEADER, payload: "home" }]);
  });
});
