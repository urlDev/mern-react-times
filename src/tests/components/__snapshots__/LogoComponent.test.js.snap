// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LogoComponent Should match the snapshot 1`] = `
<Styled(Link)
  to="/home"
>
  <img
    alt="react times logo"
    onClick={[Function]}
    style={
      Object {
        "height": "16px",
      }
    }
  />
</Styled(Link)>
`;
